### PR TITLE
Misternu3

### PR DESCRIPTION
**Pull request by misternu**

### CODE REVIEW

**Comment by misternu** [comment]
@mattbaker

File: game.exs [hunk 1]
```diff
@@ -1,7 +1,39 @@
...
+
+  def get_guess() do
+    IO.gets("Guess a number: ")
+      |> String.trim
+      |> String.to_integer
+  end
+
+  def play(), do: main(Enum.random(1..100), 7)
+
+  def main(number, turns) do
```
Comment (mattbaker) [inline game.exs:26]: what if you pattern matched in main? `def main(number, 0)` and `def main(number, turns)`?

File: game.exs [hunk 2]
```diff
@@ -1,7 +1,39 @@
...
+    print_turns(turns)
+    if (turns > 0) do
```
Comment (mattbaker) [inline game.exs:28]: No need for this `if` if you define separate `main` functions.

File: game.exs [hunk 3]
```diff
@@ -1,7 +1,39 @@
...
-    IO.puts("hello world")
+  def compare(n, t) do
+    cond do
+      n < t -> :low
+      n == t -> :equal
+      n > t -> :high
+    end
+  end
+
+  def print_state(:low),      do: IO.puts("Too low")
```
Comment (mattbaker) [inline game.exs:11]: I like the name `print_result` or `print_outcome` more, you're not really printing state, you're printing the result of the guess.

File: game.exs [hunk 4]
```diff
@@ -1,7 +1,39 @@
...
+  def print_state(:equal),    do: IO.puts("That's the right number, you win!")
```
Comment (mattbaker) [inline game.exs:12]: I'd actually keep `case` for `print_state`, I just wanted to point out that you had options in my last review :)

**Review by mattbaker — CHANGES_REQUESTED** [review]

File: game.exs (hunk 2 above)
Comment (misternu) [reply game.exs:28]: I had some sort of block stopping me from seeing that I could use pattern matching with a function that sometimes recurses, ty sensei

**Pull request closed without merging**